checkov/ansible/graph_builder/local_graph.py

```diff
@@ -0,0 +1,92 @@
+from checkov.common.graph.graph_builder.graph_components.blocks import Block
+from checkov.common.runners.graph_builder.local_graph import ObjectLocalGraph
+from checkov.common.util.consts import START_LINE, END_LINE
+from checkov.ansible.graph_builder.graph_components.resource_types import ResourceType
+from checkov.ansible.utils import get_scannable_file_paths, TASK_RESERVED_KEYWORDS, parse_file
+
+
+class AnsibleLocalGraph(ObjectLocalGraph):
+    def __init__(self, definitions: dict[str | Path, dict[str, Any] | list[dict[str, Any]]]) -> None:
+        super().__init__(definitions=definitions)
+
+        self.source = "Ansible"
+
+    def _create_vertices(self) -> None:
+        for file_path, definition in self.definitions.items():
+            if not isinstance(definition, list):
+                logging.warning(f"definition of file {file_path} has the wrong type {type(definition)}")
+                return
+
+            file_path = str(file_path)
+
+            for code_block in definition:
+                if ResourceType.TASKS in code_block:
+                    for task in code_block[ResourceType.TASKS]:
+                        self._create_tasks_vertices(file_path=file_path, task=task)
+                else:
+                    self._create_tasks_vertices(file_path=file_path, task=code_block)
+
+    def _create_tasks_vertices(self, file_path: str, task: Any) -> None:
+        """Creates tasks vertices"""
+
+        if not task or not isinstance(task, dict):
+            return
+
+        # grab the task name at the beginning before trying to find the actual module name
+        task_name = task.get("name") or "unknown"
+
+        for name, config in task.items():
+            if name in TASK_RESERVED_KEYWORDS:
+                continue
+            if name in (START_LINE, END_LINE):
+                continue
+
+            resource_type = f"{ResourceType.TASKS}.{name}"
+            block_name = f"{resource_type}.{task_name}"
+
+            attributes = deepcopy(config)
+            attributes[CustomAttributes.RESOURCE_TYPE] = resource_type
+
+            # only the module code is relevant for validation,
+            # but in the check result the whole task should be visible
+            attributes[START_LINE] = task[START_LINE]
+            attributes[END_LINE] = task[END_LINE]
+
+            self.vertices.append(
+                Block(
+                    name=block_name,
+                    config=config,
+                    path=file_path,
+                    block_type=BlockType.RESOURCE,
+                    attributes=attributes,
+                    id=block_name,
+                    source=self.source,
+                )
+            )
+
+            # no need to further check
+            break
+
+    def _create_edges(self) -> None:
+        return None
+
+    @staticmethod
+    def get_files_definitions(root_folder: str | Path) -> dict[str | Path, dict[str, Any] | list[dict[str, Any]]]:
```

Comment:
I feel like this logic is better fitted into the parser/runner, no?

Reply:
yeah, this was a solution I added, because I couldn't find a better way yet. I added a comment in the parent class 😄 
```python
    @staticmethod
    @abstractmethod
    def get_files_definitions(root_folder: str | Path) -> dict[str | Path, dict[str, Any] | list[dict[str, Any]]]:
        """This is temporary till I have a better idea"""
        pass
```